Neos.Neos/Classes/Service/ImageVariantGarbageCollector.php

```diff
@@ -62,24 +62,24 @@ public function removeUnusedImageVariant(Node $node, $propertyName, $oldValue, $
         if ($oldValue === $value || (!$oldValue instanceof ImageVariant)) {
             return;
         }
-        $identifier = $this->persistenceManager->getIdentifierByObject($oldValue);
-        $usage = $this->assetUsageRepository->findUsages(AssetUsageFilter::create()->withAsset($identifier));
+        $usageCount = $this->assetService->getUsageCount($oldValue);
 
         // This case shouldn't happen as the query will usually find at least the node that triggered this call,
         // still if there is no relation we can remove the ImageVariant.
-        if ($usage->count() === 0) {
+        if ($usageCount === 0) {
             $this->assetRepository->remove($oldValue);
             return;
         }
 
-        if ($usage->count() === 1) {
-            foreach ($usage->getIterator() as $usageItem) {
+        if ($usageCount === 1) {
+            foreach ($this->assetService->getUsageReferences($oldValue) as $usageItem) {
                 // If the result contains exactly the node that got a new ImageVariant assigned
                 // then we are safe to remove the asset here.
                 if (
-                    $usageItem->contentStreamIdentifier === $node->subgraphIdentity->contentStreamId
-                    && $usageItem->originDimensionSpacePoint === $node->originDimensionSpacePoint->hash
-                    && $usageItem->nodeAggregateIdentifier === $node->nodeAggregateId
+                    $usageItem instanceof ContentRepositoryNodeInformationInterface
+                    && $usageItem->getContentStreamId() === $node->subgraphIdentity->contentStreamId
```

Comment:
Careful, two Content Stream VOs share the same instance but we should never rely on that fact
This should either be
```suggestion
                    && $usageItem->getContentStreamId()->equals($node->subgraphIdentity->contentStreamId)
```
or
```suggestion
                    && $usageItem->getContentStreamId()->value === $node->subgraphIdentity->contentStreamId->value
```

Reply:
Was already there, but is fixed now.